src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/DataAnnotationsModelMetadataProvider.cs

```diff
@@ -17,11 +23,21 @@ public class DataAnnotationsModelMetadataProvider : AssociatedMetadataProvider<C
             return new CachedDataAnnotationsModelMetadata(this, containerType, modelType, propertyName, attributes);
         }
 
+        /// <inheritdoc />
+        /// <remarks>
+        /// Copies only a few values from the <paramref name="prototype"/>. Unlikely the rest have been computed.
+        /// </remarks>
         protected override CachedDataAnnotationsModelMetadata CreateMetadataFromPrototype(
             CachedDataAnnotationsModelMetadata prototype,
             Func<object> modelAccessor)
         {
-            return new CachedDataAnnotationsModelMetadata(prototype, modelAccessor);
+            var metadata = new CachedDataAnnotationsModelMetadata(prototype, modelAccessor);
+            foreach (var keyValuePair in prototype.AdditionalValues)
+            {
+                metadata.AdditionalValues.Add(keyValuePair);
+            }
```

Comment:
might want to consider whether this belongs here or in the constructor of one of these classes. The thought occurred to me but I don't know what's best for the layering here. I think I trust whatever you decide.